### DISCUSSION TITLE
Tracing variable "f" of the forward_once() code

### DISCUSSION THREAD

**Discussion by hanming-nls** [ultralytics/yolov5#4859] [Q&A] [answered]
Pardon me as I'm relatively new to Python.

As I'm tracing through the forward_once() code (line 141) in yolo.py, it checks if the module if from the previous layer by checking the class variable "f" is having the value -1.
```
for m in self.model:
    if m.f != -1:  # if not from previous layer
        x = y[m.f] if isinstance(m.f, int) else [x if j == -1 else y[j] for j in m.f]  # from earlier layers
```
It seems that for the layer Concat and Detect, it will return a len(3) array while the others return -1.

My question is may I know where is this variable "f" defined?

As these are all sub-classes of Module which in turns is  sub-class of Object, the self.f should be defined in either of these classes.
However, I just can't seems to find the declaration.
Is it dynamically defined?

**Comment by glenn-jocher** [comment, marked as answer]
@hanming-nls this is easy to find if you just search the same file for all instances of `.f`:

https://github.com/ultralytics/yolov5/blob/9ef94940aa5e9618e7e804f0758f9a6cebfc63a9/models/yolo.py#L267

**Reply by hanming-nls** [reply]
Thanks!!!

My bad, I search for "self.f" "f =" and match results that are wrong.

Thanks again.  👍
